File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
'''
짝지어 제거하기
같은 알파벳이 2개 붙어있는 짝 찾아서 제거
스택 이용
'''

def solution(s):
    stack = []
    stack.append(s[0])
    for i in range(1,len(s)):
        stack.append(s[i])
        if((len(stack) >= 2) and (stack[-1] == stack[-2])):
            stack.pop()
            stack.pop()
    if(len(stack) == 0):
        return 1
    else:
        return 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;
class Solution
{
    public int solution(String s)
    {
        Stack<Character> stack = new Stack<>();
        stack.add(s.charAt(0));
        for(int i = 1; i < s.length(); i++){
            stack.add(s.charAt(i));
            if(stack.size() >= 2 && (stack.get(stack.size()-1) == stack.get(stack.size()-2))){
                stack.pop();
                stack.pop();
            }
        }
        if(stack.size() == 0) return 1;
        else return 0;
    }
}